### ISSUE TITLE
Kibana server is not ready yet: current license is non-compliant for [security]

### ISSUE THREAD

**Issue by murilolivorato** [deviantony/docker-elk#899]
hello ,  I was working before  , but now it is giving this error -

Kibana server is not ready yet.

```
docker-compose logs


| {"@timestamp":"2023-08-18T15:57:57.743Z", "log.level":"ERROR", "message":"collector [cluster_stats] failed to collect data", "ecs.version": "1.2.0","service.name":"ES_ECS","event.dataset":"elasticsearch.server","process.thread.name":"elasticsearch[elasticsearch][generic][T#8]","log.logger":"org.elasticsearch.xpack.monitoring.collector.cluster.ClusterStatsCollector","elasticsearch.cluster.uuid":"qyWikjrMQc2IPR1bt44hjg","elasticsearch.node.id":"d8DCCAzDQGCLFPA7rX5QSQ","elasticsearch.node.name":"elasticsearch","elasticsearch.cluster.name":"docker-cluster","error.type":"org.elasticsearch.ElasticsearchSecurityException","error.message":"current license is non-compliant for [security]","error.stack_trace":"org.elasticsearch.ElasticsearchSecurityException: current license is non-compliant for [security]\n\tat org.elasticsearch.xcore@8.8.2/org.elasticsearch.license.LicenseUtils.newComplianceException(LicenseUtils.java:40)\n\tat org.elasticsearch.security@8.8.2/org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.apply(SecurityActionFilter.java:94)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:84)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:61)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.tasks.TaskManager.registerAndExecute(TaskManager.java:199)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.client.internal.node.NodeClient.executeLocally(NodeClient.java:112)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.client.internal.node.NodeClient.doExecute(NodeClient.java:90)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:379)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:365)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.client.internal.support.AbstractClient$ClusterAdmin.execute(AbstractClient.java:666)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:34)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.action.ActionRequestBuilder.get(ActionRequestBuilder.java:41)\n\tat org.elasticsearch.xpack.monitoring.collector.cluster.ClusterStatsCollector.doCollect(ClusterStatsCollector.java:90)\n\tat org.elasticsearch.xpack.monitoring.collector.Collector.collect(Collector.java:94)\n\tat org.elasticsearch.xpack.monitoring.MonitoringService$MonitoringExecution$1.doRun(MonitoringService.java:269)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)\n\tat org.elasticsearch.server@8.8.2/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:916)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1623)\n"}


```

```
Docker Compose version 2.19.1
Docker version 24.0.2, build cb74dfcd85
```

**Comment by antoineco** [comment]
It seems like your "trial" license expired and you tried to switch it to "basic" inside the `elasticsearch.yml` configuration file, which is not how licenses are updated in Elasticsearch.

Please check https://github.com/deviantony/docker-elk#how-to-disable-paid-features

---

*tl;dr*: switch your license like this

```sh
curl -D- -X POST -u elastic:changeme 'http://localhost:9200/_license/start_basic?acknowledge=true'
```
